### PR TITLE
Boosting Performance and Efficiency with Concurrent Async Task Execution Using `Promise.all` ✈

### PR DESCRIPTION
**Pull request by sanjaiyan-dev**
Hi, this PR aims to improve performance by concurrently running async tasks using `Promise.all`. Here are some benefits of doing so:

1. **Faster Execution:** Concurrent execution of async tasks can significantly reduce the overall execution time, as tasks that don't depend on each other can run in parallel.

2. **Optimal Resource Utilization:** Utilizing `Promise.all` allows you to make better use of available system resources, as it doesn't block the main thread and enables more efficient resource allocation.

Extremely sorry if I made any mistakes :(

### CODE REVIEW

**Comment by sanjaiyan-dev** [comment]
Sorry, I am not able to run the `rush format` command; otherwise, I think everything is fine 🤞🏽

File: packages/compiler/src/init/init.ts [hunk 1]
```diff
@@ -153,8 +153,10 @@ export async function initTypeSpecProject(
     },
   ]);
 
-  const libraries = await selectLibraries(template);
-  const parameters = await promptCustomParameters(template);
+  const [libraries, parameters] = await Promise.all([
```
Comment (timotheeguerin) [inline packages/compiler/src/init/init.ts:156]: This one we do need fore sure to be done in sequence it prompts the user.

**Review by timotheeguerin — COMMENTED** [review]
Hi @sanjaiyan-dev thanks a lot for the contribution, however from looking at those it seems most are in places where performance is not really an issue and running in parallel always makes debugging and logging much harder so I would probably hold off. 
We can see what other think for this.

File: packages/compiler/src/init/init.ts (hunk 1 above)
Comment (sanjaiyan-dev) [reply packages/compiler/src/init/init.ts]: > This one we do need fore sure to be done in sequence it prompts the user.

Done 🤝

**Comment by sanjaiyan-dev** [comment]
> Hi @sanjaiyan-dev thanks a lot for the contribution, however from looking at those it seems most are in places where performance is not really an issue and running in parallel always makes debugging and logging much harder so I would probably hold off. We can see what other think for this.

Ok :)

**Comment by timotheeguerin** [comment]
H @sanjaiyan-dev sorry for the delay lost track of this, thanks again for trying to contribute, as my previous message I don't think this is really worth it to run in parallel in those places, I'll close this PR for now

**Pull request closed without merging**